Maths/AverageMedian.js

```diff
@@ -9,22 +9,14 @@
  */
 
 const averageMedian = (sourceArrayOfNumbers) => {
-  let numbers = [...sourceArrayOfNumbers]
-  let median = 0
+  const numbers = [...sourceArrayOfNumbers].sort(sortNumbers)
   const numLength = numbers.length
-  numbers = numbers.sort(sortNumbers)
 
   if (numLength % 2 === 0) {
-    median = (numbers[numLength / 2 - 1] + numbers[numLength / 2]) / 2
-  } else {
-    median = numbers[(numLength - 1) / 2]
-  }
-
-  return median
+    return (numbers[numLength / 2 - 1] + numbers[numLength / 2]) / 2
+  } else return numbers[Math.floor(numLength / 2)]
```

Comment:
What about `numLength >> 1`?

Reply:
let me try that.

Reply:
it wont work let me explain with example 
after sort array looks like this `[1, 2, 5, 8, 9, 10, 11]`
as its length is 7 and in binary it will be 111 
shifting it two positions gives us 1 and median is at 4th position in sorted array